File: sol/solution/0001-0100/0013. Roman to Integer/Solution.scala
object Solution {
    def romanToInt(s: String): Int = {
        val map = Map('I' -> 1, 'V' -> 5, 'X' -> 10, 'L' -> 50, 'C' -> 100, 'D' -> 500, 'M' -> 1000)
        s.indices.foldLeft(0) { (res, i) =>
            if (i < s.length - 1 && map(s(i)) < map(s(i + 1))) res - map(s(i))
            else res + map(s(i))
        }
    }
}

File: sol/solution/0001-0100/0013. Roman to Integer/Solution.rs
impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let map = |c| match c {'I' => 1, 'V' => 5, 'X' => 10, 'L' => 50, 'C' => 100, 'D' => 500, 'M' => 1000, _ => 0};
        let chars: Vec<char> = s.chars().collect();
        let mut res = 0;
        for i in 0..chars.len() {
            let val = map(chars[i]);
            if i + 1 < chars.len() && val < map(chars[i + 1]) {
                res -= val;
            } else {
                res += val;
            }
        }
        res
    }
}